src/heartbeat/heartbeat.c

```diff
@@ -196,45 +209,71 @@ void hb_tx_cb(uint8_t* data, uint8_t* len) {
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_OBC;
+            data[OP_CODE] = PING_REQUEST;
         } else if (hb_send_eps_ping) {
-            data[0] = 1;
-            data[1] = hb_self_id;
-            data[2] = HB_EPS;
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_EPS;
+            data[OP_CODE] = PING_REQUEST;
         } else if (hb_send_pay_ping) {
-            data[0] = 1;
-            data[1] = hb_self_id;
-            data[2] = HB_PAY;
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_PAY;
+            data[OP_CODE] = PING_REQUEST;
         } else if (hb_send_obc_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_OBC;
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_OBC;
+            data[OP_CODE] = PING_RESPONSE;
         } else if (hb_send_eps_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_EPS;
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_EPS;
+            data[OP_CODE] = PING_RESPONSE;
         } else if (hb_send_pay_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_PAY;
-        } else {
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_PAY;
+            data[OP_CODE] = PING_RESPONSE;
+        } else if (hb_send_obc_rdata_resp){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_OBC;
+            data[OP_CODE] = RESTART_DATA_REQ;
+        } else if (hb_send_eps_rdata_resp){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_EPS;
+            data[OP_CODE] = RESTART_DATA_REQ;
+        } else if (hb_send_pay_rdata_resp){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_PAY;
+            data[OP_CODE] = RESTART_DATA_REQ;
+        } else if (hb_received_obc_rdata_req){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_OBC;
+            data[OP_CODE] = RESTART_DATA_RESP;
+            data[RESTART_REASON] = eeprom_read_byte((uint8_t*)RESTART_REASON_EEPROM_ADDR);
+            data[RESTART_COUNT] = eeprom_read_byte((uint8_t*)RESTART_COUNT_EEPROM_ADDR);
+        } else if (hb_received_eps_rdata_req){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_EPS;
+            data[OP_CODE] = RESTART_DATA_RESP;
+            data[RESTART_REASON] = eeprom_read_byte((uint8_t*)RESTART_REASON_EEPROM_ADDR);
+            data[RESTART_COUNT] = eeprom_read_byte((uint8_t*)RESTART_COUNT_EEPROM_ADDR);
+        } else if (hb_received_pay_rdata_req){
+            data[SENDER] = hb_self_id;
+            data[RECEIVER] = HB_PAY;
+            data[OP_CODE] = RESTART_DATA_RESP;
+            data[RESTART_REASON] = eeprom_read_byte((uint8_t*)RESTART_REASON_EEPROM_ADDR);
+            data[RESTART_COUNT] = eeprom_read_byte((uint8_t*)RESTART_COUNT_EEPROM_ADDR);
+        }
```

Comment:
1. Restart count is 4 bytes instead of 1 byte, so you need to account for this
2. You don't need to read from EEPROM, just directly access the `restart_reason` and `restart_count` variables from `uptime.c`

Reply:
I'll make these changes tomorrow, but I have a couple questions.

1. As the reset count will foreseeably not exceed an 8-bit unsigned integer, is there not any reason to leave bits 5-7 empty should we choose to use them for something else in the future?
2. I noticed that in the uptime.c file that the restart reason is written/read as a 32 bit int. Should we reduce this to 8 bits, or assume that masking it to 8 bits in heartbeat is sufficient?